src/test/java/com/commercetools/sync/commons/utils/CustomUpdateActionUtilsTest.java

```diff
@@ -546,6 +546,22 @@ void buildNewOrModifiedCustomFieldsUpdateActions_WithNoNewOrModifiedCustomFields
         assertThat(updateActions).isEmpty();
     }
 
+    @Test
+    void buildNewOrModifiedCustomFieldsUpdateActions_WithNullCustomFields_ShouldNotBuildActions() {
+        final Map<String, JsonNode> oldCustomFields = new HashMap<>();
+        oldCustomFields.put("invisibleInShop", JsonNodeFactory.instance.booleanNode(true));
+        oldCustomFields.put("backgroundColor", JsonNodeFactory.instance.objectNode().put("de", "rot").put("en", "red"));
+
+        final Map<String, JsonNode> newCustomFields = new HashMap<>();
+        newCustomFields.put("invisibleInShop", null);
+
+        final List<UpdateAction<Product>> updateActions = buildNewOrModifiedCustomFieldsUpdateActions(oldCustomFields,
+            newCustomFields, mock(Price.class), new PriceCustomActionBuilder(), 1, Price::getId);
+
+        assertThat(updateActions).isNotNull();
+        assertThat(updateActions).isEmpty();
+    }
+
```

Comment:
I would also add these unit tests:

```java
    @Test
    void buildSetCustomFieldsUpdateActions_WithNullAsValue_ShouldCorrectlyBuildAction() {
        // preparation
        final Map<String, JsonNode> oldCustomFieldsMap = new HashMap<>();
        oldCustomFieldsMap.put("setOfBooleans", JsonNodeFactory
            .instance
            .arrayNode()
            .add(JsonNodeFactory.instance.booleanNode(false)));

        final CustomFields oldCustomFields = mock(CustomFields.class);
        when(oldCustomFields.getFieldsJsonMap()).thenReturn(oldCustomFieldsMap);

        final Asset oldAsset = mock(Asset.class);
        when(oldAsset.getCustom()).thenReturn(oldCustomFields);


        final Map<String, JsonNode> newCustomFieldsMap = new HashMap<>();
        newCustomFieldsMap.put("setOfBooleans", null);


        // test
        final List<UpdateAction<Product>> updateActions =
            buildSetCustomFieldsUpdateActions(oldCustomFieldsMap, newCustomFieldsMap, mock(Asset.class), new AssetCustomActionBuilder(), 1, Asset::getId);

        // assertion
        assertThat(updateActions)
            .containsExactly(SetAssetCustomField
                .ofVariantIdUsingJsonAndAssetKey(1, oldAsset.getKey(), "setOfBooleans", null, true));
    }

    @Test
    void buildSetCustomFieldsUpdateActions_WithNullJsonNodeAsValue_ShouldCorrectlyBuildAction() {
        // preparation
        final Map<String, JsonNode> oldCustomFieldsMap = new HashMap<>();
        oldCustomFieldsMap.put("setOfBooleans", JsonNodeFactory
            .instance
            .arrayNode()
            .add(JsonNodeFactory.instance.booleanNode(false)));

        final CustomFields oldCustomFields = mock(CustomFields.class);
        when(oldCustomFields.getFieldsJsonMap()).thenReturn(oldCustomFieldsMap);

        final Asset oldAsset = mock(Asset.class);
        when(oldAsset.getCustom()).thenReturn(oldCustomFields);


        final Map<String, JsonNode> newCustomFieldsMap = new HashMap<>();
        newCustomFieldsMap.put("setOfBooleans", JsonNodeFactory.instance.nullNode());


        // test
        final List<UpdateAction<Product>> updateActions =
            buildSetCustomFieldsUpdateActions(oldCustomFieldsMap, newCustomFieldsMap, mock(Asset.class), new AssetCustomActionBuilder(), 1, Asset::getId);

        // assertion
        assertThat(updateActions)
            .containsExactly(SetAssetCustomField
                .ofVariantIdUsingJsonAndAssetKey(1, oldAsset.getKey(), "setOfBooleans", JsonNodeFactory.instance.nullNode(), true));
    }
```

Reply:
I would also add these 2 unit tests in `ProductVariantAssetUpdateActionUtilsTest`:


```java
    @Test
    void buildCustomUpdateActions_WithNullValue_ShouldCorrectlyBuildAction() {
        // preparation
        final Map<String, JsonNode> oldCustomFieldsMap = new HashMap<>();
        oldCustomFieldsMap.put("invisibleInShop", JsonNodeFactory.instance.booleanNode(true));
        oldCustomFieldsMap.put("setOfBooleans", JsonNodeFactory
            .instance
            .arrayNode()
            .add(JsonNodeFactory.instance.booleanNode(false)));

        final Map<String, JsonNode> newCustomFieldsMap = new HashMap<>();
        newCustomFieldsMap.put("invisibleInShop", JsonNodeFactory.instance.booleanNode(true));
        newCustomFieldsMap.put("setOfBooleans", null);


        final CustomFields oldCustomFields = mock(CustomFields.class);
        final String typeId = UUID.randomUUID().toString();
        when(oldCustomFields.getType()).thenReturn(Type.referenceOfId(typeId));
        when(oldCustomFields.getFieldsJsonMap()).thenReturn(oldCustomFieldsMap);

        final CustomFieldsDraft newCustomFieldsDraft =
            CustomFieldsDraft.ofTypeIdAndJson(typeId, newCustomFieldsMap);

        final Asset oldAsset = mock(Asset.class);
        when(oldAsset.getCustom()).thenReturn(oldCustomFields);

        final AssetDraft newAssetDraft = AssetDraftBuilder.of(emptyList(), empty())
                                                          .custom(newCustomFieldsDraft)
                                                          .build();

        final List<String> errors = new ArrayList<>();

        final ProductSyncOptions syncOptions = ProductSyncOptionsBuilder.of(mock(SphereClient.class))
                                                                        .errorCallback((errorMessage, throwable) ->
                                                                            errors.add(errorMessage))
                                                                        .build();
        // test
        final List<UpdateAction<Product>> updateActions =
            buildCustomUpdateActions(1, oldAsset, newAssetDraft, syncOptions);

        // assertion
        assertThat(errors).isEmpty();
        assertThat(updateActions)
            .containsExactly(SetAssetCustomField
                .ofVariantIdUsingJsonAndAssetKey(1, oldAsset.getKey(), "setOfBooleans", null, true));
    }

    @Test
    void buildCustomUpdateActions_WithNullJsonNodeValue_ShouldCorrectlyBuildAction() {
        // preparation
        final Map<String, JsonNode> oldCustomFieldsMap = new HashMap<>();
        oldCustomFieldsMap.put("setOfBooleans", JsonNodeFactory
            .instance
            .arrayNode()
            .add(JsonNodeFactory.instance.booleanNode(false)));

        final Map<String, JsonNode> newCustomFieldsMap = new HashMap<>();
        newCustomFieldsMap.put("setOfBooleans", JsonNodeFactory.instance.nullNode());


        final CustomFields oldCustomFields = mock(CustomFields.class);
        final String typeId = UUID.randomUUID().toString();
        when(oldCustomFields.getType()).thenReturn(Type.referenceOfId(typeId));
        when(oldCustomFields.getFieldsJsonMap()).thenReturn(oldCustomFieldsMap);

        final CustomFieldsDraft newCustomFieldsDraft =
            CustomFieldsDraft.ofTypeIdAndJson(typeId, newCustomFieldsMap);

        final Asset oldAsset = mock(Asset.class);
        when(oldAsset.getCustom()).thenReturn(oldCustomFields);

        final AssetDraft newAssetDraft = AssetDraftBuilder.of(emptyList(), empty())
                                                          .custom(newCustomFieldsDraft)
                                                          .build();

        final List<String> errors = new ArrayList<>();

        final ProductSyncOptions syncOptions = ProductSyncOptionsBuilder.of(mock(SphereClient.class))
                                                                        .errorCallback((errorMessage, throwable) ->
                                                                            errors.add(errorMessage))
                                                                        .build();
        // test
        final List<UpdateAction<Product>> updateActions =
            buildCustomUpdateActions(1, oldAsset, newAssetDraft, syncOptions);

        // assertion
        assertThat(errors).isEmpty();
        assertThat(updateActions)
            .containsExactly(SetAssetCustomField
                .ofVariantIdUsingJsonAndAssetKey(1, oldAsset.getKey(), "setOfBooleans", JsonNodeFactory.instance.nullNode(), true));
    }
```

Reply:
89aab968ca658892ef582d74b6a2a5644592eb63